src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,180 @@
+    privateChats: state.firestore.ordered.privatechats,
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const lastMessageByUserReducer = useCallback(
+    (agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    },
+    [user]
+  );
+
+  const discussionPartnerWithLastMessageExchanged = useMemo(() => {
+    if (!privateChats) return {};
+    return privateChats.reduce<LastMessageByUser>(lastMessageByUserReducer, {});
+  }, [lastMessageByUserReducer, privateChats]);
+
+  const onClickOnSender = useCallback(
+    (sender: WithId<User>) => {
+      const chatsToUpdate = privateChats.filter(
+        (chat) => !chat.isRead && chat.from === sender.id
+      );
+      chatsToUpdate.map(
+        (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+      );
+      setSelectedUser(sender);
+    },
+    [privateChats, user]
+  );
+
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = useCallback(
+    async (data: { messageToTheBand: string }) => {
+      chatContext &&
+        user &&
+        chatContext.sendPrivateChat(
+          user.uid,
+          selectedUser!.id,
+          data.messageToTheBand
+        );
+    },
+    [chatContext, selectedUser, user]
+  );
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  const discussions = useMemo(() => {
+    return Object.keys(discussionPartnerWithLastMessageExchanged).sort((a, b) =>
+      discussionPartnerWithLastMessageExchanged[b].ts_utc
+        .valueOf()
+        .localeCompare(
+          discussionPartnerWithLastMessageExchanged[a].ts_utc.valueOf()
+        )
+    );
+  }, [discussionPartnerWithLastMessageExchanged]);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
```

Comment:
This will cause re-renders due to the wrapping function reference (`() =>`) changing each render

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044